ParquetClassLibrary/Crafting/StrikePanel.cs

```diff
@@ -0,0 +1,48 @@
+using System;
+using ParquetClassLibrary.Utilities;
+
+namespace ParquetClassLibrary.Crafting
+{
+    /// <summary>
+    /// Models the panels that the player must strike during item crafting.
+    /// </summary>
+    public struct StrikePanel
+    {
+        /// <summary><c>true</c> if this panel is not used in the current crafting recipe; otherwise, <c>false</c>.</summary>
+        public bool IsVoid { get; set; }
+
+        /// <summary>
+        /// The range of values this panel can take on while being worked.  <see cref="Range{T}.Minimum"/> is normally 0.
+        /// </summary>
+        public Range<int> WorkingRange { get; set; }
+
+        /// <summary>Backing value for <see cref="IdealRange"/>.</summary>
+        private Range<int> _iealRange;
+
+        /// <summary>
+        /// The range of values this panel targets for a completed craft.
+        /// The values must fall within the range of <c see="WorkingRange"/>.
+        /// </summary>
+        /// <exception cref="ArgumentOutOfRangeException">
+        /// Thrown when the either the <see cref="Range{T}.Maximum"/> or <see cref="Range{T}.Minimum"/> are beyond
+        /// the range specified in <c see="WorkingRange"/>.
+        /// </exception>
+        public Range<int> IdealRange
+        {
+            get
+            {
+                return _iealRange;
+            }
+            set
+            {
+                if (value.Maximum > WorkingRange.Maximum ||
```

Comment:
If this StrikePanel is Void, what should you set IdealRange to?

Reply:
Hmm, so, if StrikePanel is void then the ranges should just be ignored.  It doesn't matter at that point what they are set to.  I don't want to wipe out any set values, though, as it may be advantageous for a panel editor to preserve values in case someone hits the Void checkbox by accident.

What do you think?

Reply:
Yup! That acts like how I'd want it to, I think.